### PR TITLE
enh(UI): "Simplify" metrics selection 

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This improves the metrics selection by allowing a fine-grained selection on metrics and resources related.

https://github.com/centreon/centreon/assets/12515407/6d59b79e-3030-4981-b225-b9e72872e414

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by kduret — APPROVED** [review]
ok for cypress e2e changes

**Review by kduret — DISMISSED** [review]

File: centreon/packages/ui/src/components/CollapsibleItem/CollapsibleItem.cypress.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const customizedTitle = <div>Customized title</div>;
+
+const initialize = (props: Omit<Props, 'children'>): void => {
+  cy.mount({
+    Component: <CollapsibleItem {...props}>Content</CollapsibleItem>
+  });
+};
+
+describe('CollapsibleItem', () => {
+  it('displays the component collpased by default', () => {
```
Comment (bdauria) [inline centreon/packages/ui/src/components/CollapsibleItem/CollapsibleItem.cypress.spec.tsx:14]:
```suggestion
  it('displays the component collapsed by default', () => {
```

File: centreon/packages/ui/src/components/ItemComposition/ItemComposition.cypress.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  it('displays add button as hidden when the prop is set to true', () => {
+    initialize({ addbuttonDisabled: true });
+
+    cy.findByTestId('Add').should('be.disabled');
+
+    cy.makeSnapshot();
+  });
+
+  it('does not display the add button as hidden when the prop is set to true', () => {
```
Comment (bdauria) [inline centreon/packages/ui/src/components/ItemComposition/ItemComposition.cypress.spec.tsx:80]:
```suggestion
  it('does not display the add button when the prop is set to true', () => {
```

File: centreon/packages/ui/src/components/ItemComposition/ItemComposition.cypress.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    initialize({ addButtonHidden: true });
+
+    cy.findByTestId('Add').should('not.exist');
+
+    cy.makeSnapshot();
+  });
+
+  it('does not display the delete button as hidden when the prop is set to true', () => {
```
Comment (bdauria) [inline centreon/packages/ui/src/components/ItemComposition/ItemComposition.cypress.spec.tsx:88]:
```suggestion
  it('does not display the delete button when the prop is set to true', () => {
```

**Review by bdauria — APPROVED** [review]

**Review by MedWassCh — APPROVED** [review]

**Pull request merged**